x/streamer/keeper/abci.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/dymensionxyz/dymension/v3/x/streamer/types"
+)
+
+func (k Keeper) EndBlock(ctx sdk.Context) error {
+	streams := k.GetActiveStreams(ctx)
+
+	epochPointers, err := k.GetAllEpochPointers(ctx)
+	if err != nil {
+		return fmt.Errorf("get all epoch pointers: %w", err)
+	}
+
+	maxIterations := k.GetParams(ctx).MaxIterationsPerBlock
+	totalIterations := uint64(0)
+	totalDistributed := sdk.NewCoins()
+
+	for _, p := range epochPointers {
```

Comment:
is it sorted? I guess it make more sense to start with the short epochs (e.g min before day)